metabase/metabase
# Add routes for model detail page tabs

Epic #27581

Adds dedicated route per each model detail page's tab:

* Used by — `/model/:slug/detail/usage`
* Schema — `/model/:slug/detail/schema`
* Actions — `/model/:slug/detail/actions`

### To Verify

1. Open  `/model/:id/detail` (ensure tab isn't specified)
2. Ensure you see the "Used by" tab by default
3. Click through tabs, ensure you can see the expected content and that URLs match the ones listed above
4. Create a model based on data from a database that doesn't have actions enabled (can be turned off at admin db page) or that doesn't support actions at all (any database except Postgres and MySQL)
5. Try to open `/model/:id/detail/actions` for this model and ensure you're navigated to `/usage` instead
6. Try to open a tab that doesn't exist, e.g. `/model/:id/detail/foo-bar` and ensure you're navigated to `/usage`


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/27959)
